flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/java/BatchTableEnvironment.java

```diff
@@ -105,9 +106,30 @@
 	 * @param fields The field names of the resulting {@link Table}.
 	 * @param <T> The type of the {@link DataSet}.
 	 * @return The converted {@link Table}.
+	 * @deprecated use {@link #fromDataSet(DataSet, Expression...)}
 	 */
+	@Deprecated
 	<T> Table fromDataSet(DataSet<T> dataSet, String fields);
 
+	/**
+	 * Converts the given {@link DataSet} into a {@link Table} with specified field names.
+	 *
+	 * Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   DataSet<Tuple2<String, Long>> set = ...
+	 *   Table tab = tableEnv.fromDataSet(set, $("a"), $("b").as("name"), $("timestamp").rowtime());
```

Comment:
the example makes no sense. Let's use a tuple 3 and explain each expression.
```
DataSet<Tuple3<String, Long, Timestamp>> set = ...
Table tab = tableEnv.fromDataSet(
  set,
  $("f0"), // use the original field
  $("f1").as("name"), // give the original field a better name
  $("f2").as("timestamp").rowtime() // give the original field a better name and declare it as a rowtime attribute
);
```

Reply:
please verify if my code is correct.

Reply:
I am really sorry for that... I naively copied the javadoc over from StreamTableEnvironment. Forgot that in DataSet the timestamp comes from the record.